lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -694,14 +770,37 @@ Inliner::mapPortsToWires(StringRef prefix, OpBuilder &b,
       newAnnotations.push_back(anno.getAttr());
     }
 
-    auto wire = b.create<WireOp>(
-        target.getLoc(), type, (prefix + portInfo[i].getName()).str(),
-        NameKindEnum::DroppableName, ArrayAttr::get(context, newAnnotations),
-        newSym);
+    Value wire =
+        TypeSwitch<FIRRTLType, Value>(type)
```

Comment:
In a followup PR, maybe we can use the same strategy to avoid any temporary wires, not just `RefType`.

Reply:
Hopefully! And maybe roll our own backedgebuilder-ish or so -- plumbing the edge list is redundant with the list it already keeps internally, for example.

> maybe we can use the same strategy to avoid any temporary wires

I started down that road, but am not sure we can be confident each port/result will only have a single driver (with RefType this is illegal), meaning we'll still need a wire sometimes, so maybe it's okay to just let canonicalizers drop uninteresting single-driver wires if introduced (since they'll do the same sort of checking-for-single-driver checks anyway, possibly in parallel?).